tests/src/python/test_qgstaskmanager.py

```diff
@@ -261,6 +262,36 @@ def finished_multiple_value_result(e, results):
         self.assertEqual(result_value, 5)
         self.assertEqual(result_statement, 'whoo')
 
+    def testTaskFromFunctionWithSubTaskCompletedIsCalledOnce(self):  # spellok
+        """ test that when a parent task has subtasks it does emit taskCompleted only once"""
+
+        self.finished = 0
+        self.completed = 0
+
+        def _on_finished():
```

Comment:
For me I also need to change this to:

    def _on_finished(e):

Reply:
With that change it occasionally passes here too. The problem is that it randomly fails with 

```
Traceback (most recent call last):
  File "/home/xxx/dev/QGIS/tests/src/python/test_qgstaskmanager.py", line 294, in testTaskFromFunctionWithSubTaskCompletedIsCalledOnce
    self.assertEqual(self.completed, 1)
AssertionError: 2 != 1
```

or also:

```
Traceback (most recent call last):
  File "/home/ale/dev/QGIS/tests/src/python/test_qgstaskmanager.py", line 294, in testTaskFromFunctionWithSubTaskCompletedIsCalledOnce
    self.assertEqual(self.completed, 1)
AssertionError: 0 != 1
```

I wonder if we need to add an additional processEvents before checking.

A side question is how to better bubble (Python) exceptions from subtasks to the main task? All I get from the python wrapper is 'Task canceled' from https://github.com/qgis/QGIS/blob/master/python/core/additions/qgstaskwrapper.py#L50